beginner.md

```diff
@@ -0,0 +1,588 @@
+Dijit widgets instead had a property which referenced the DOM node.
+
+For example if you open your browser developer tools and in the console enter `myvar = document.getElementById('element')` and then explore
+the properties on that variable `myvar`, you'll see it's just a regular HTML element;
+if you're more inquisitive you might be able to see there are extra properties/methods on this element which is what the `delite` framework is providing.
+
+###Registering
+
+The `<blog-post>` element doesn't constitute a custom element on its own; it first needs to go through a registration process which is achieved using
+the `delite/register` module. This is analogous to the HTML specification for registering custom elements
+i.e. `document.registerElement('blog-post');`
+
+If we look at the custom element module `./BlogPost.js` we see that we register the custom element tag via the `return register(....)` method:
+
+```js
+
+define([
+	"delite/register",
+	"delite/Widget",
+	"delite/handlebars!./BlogPost/BlogPost.html",
+    "requirejs-dplugins/css!./BlogPost/css/BlogPost.css"
+], function (register, Widget, template) {
+	return register("blog-post", [HTMLElement, Widget], {
+		baseClass: "blog-post",
+		value: "",
+		template: template
+	});
+});
+
+```
+
+
+This is an important concept which sometimes isn't clear at a first glance. You can add any non-standard tag to an HTML page and the browser HTML parser
+will not complain; this is because these elements will be defined as a native
+[`HTMLUnknownElement`](http://www.whatwg.org/specs/web-apps/current-work/multipage/dom.html#htmlunknownelement).
+To create a custom element it must be **upgraded** first; this is what `delite/register` does. `delite/register` supports browsers who natively
+support `document.registerElement` and those who don't.
+
+The registration process above using `delite/register`, creates a custom element by registering the tag name `"blog-post"` as the first
+argument and then inheriting (prototyping) the `HTMLElement` native element (as well as the `"delite/Widget"` module).
+
+Elements which inherit from `HTMLElement`
+using [valid custom element names](http://www.w3.org/TR/2013/WD-custom-elements-20130514/#dfn-custom-element-name) are custom elements.
+The most basic requirement for the tag name is it **MUST** contain a dash **(-)**.
+
+In case there's any confusion, the module name created `BlogPost` bears no relation to the custom element name i.e. `blog-post`, the module `BlogPost`
+registers the custom element tag name `blog-post`.
+
+###Declarative creation of custom elements
+If we view the generated sample `./samples/BlogPost.html`, we see the following JavaScript:
+
+```js
+require(["delite/register", "blogging-package/BlogPost"], function (register) {
+    register.parse();
+});
+```
+
+Declarative widgets (those created via markup in the page) need to be parsed in order to kick off the lifecycle of creating the widget.
```

Comment:
widget instances? (i.e. the widget themselves are not declarative or programatic just their instanciation?)